### PR TITLE
Add CSF function and test script

### PR DESCRIPTION
**Pull request by domstoppable**

### CODE REVIEW

**Comment by mmagnuski** [comment]
Hi Dominic, thanks for the PR.
Could you revert the changes to questplus.py? - we'll cover these once we deal with addition of csf psychometric function.

File: test_csf.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import numpy as np
+import matplotlib.pyplot as plt
+
+import psychometric
+from quest_plus import QuestPlus
+
+def plot_qp(qp, true_params):
```
Comment (mmagnuski) [inline test_csf.py]: we can ignore qp plotting for now, in the first PR we only need to get the csf psychometric function working and plot it. Similarily to plotting weibull in the example notebook.

File: test_csf.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+t_space = np.arange(-30, -50, -2)
+c0_space = np.arange(-40, -60, -2)
+cf_space = np.arange(.8, 1.6, .2)
+param_space = [t_space, c0_space, cf_space]
+
+qp = QuestPlus(stim_space, param_space, psychometric.csf_watson_and_ahumada)
+
+true_params = [-20, -50, 1.2]
+stim_params = qp.next_stim()
```
Comment (mmagnuski) [inline test_csf.py]: with flat priors (or more precisely - no ability to set priors) calling next_stim before ever updating qp won't give you anything useful. You can start from set stimulus, frequency pair.

**Comment by domstoppable** [comment]
I've reverted quest_plus.py and changed test_csf.py to just plot the CSF. Two things:

1. Since the CSF is a 2 dimensional function, the plot is an image (such that x,y = frequency,contrast and the color of each plotted point is the probability result from the psychometric function. Is that the right way to visualize this psychometric function?
2. I had to transpose the image data before plotting to make the shape match what I was expecting (from the CSF example in Watson's Quest+ paper), but I don't understand why. Any clues?

**Comment by mmagnuski** [comment]
Thanks, this looks good. I will test it later today and clean things up if needed. Then I'll try to adapt questplus to multidim stim.

**Comment by mmagnuski** [comment]
1. Yes, heatmap/image is the right way to visualize values in 2d space.
2. Thats due to the fact that you expect frequency, contrast stim space in your function. Watson plots contrast on y axis but y axis is the first matrix dimension, so you would need contrast, frequency in the csf function. I think it is a little more intuitive this way - I'll change it later.

**Pull request merged**